### PR TITLE
feat(schema): implement the Portex template type

### PR DESCRIPTION
**Pull request by linjiX**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2b4qrc5 实现 Portex 模板类型 python 表示](https://app.clickup.com/t/2b4qrc5)

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/schema/factory.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+        """
+        ...
+
+
+class TypeFactory(Factory):
+    """The template factory for portex type.
+
+    Arguments:
+        decl: A dict indicate a portex type.
```
Comment (Lee-000) [inline graviti/schema/factory.py:130]: `A dict which indicates ...`?

File: graviti/schema/base.py [hunk 2]
```diff
@@ -44,61 +45,57 @@ def __call__(self, class_: _S) -> _S:
...
+    _empty = Parameter.empty
+
     def __init__(
-        self, key: str, required: bool, default: Any = ..., options: Optional[Iterable[Any]] = None
+        self,
+        name: str,
+        default: Any = _empty,
+        options: Optional[Iterable[Any]] = None,
+        *,
+        annotation: Any = _empty,
```
Comment (graczhual) [inline graviti/schema/base.py:66]: `annotation` needs to be added to `Arguments`

File: graviti/schema/factory.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+        """
+        type_kwargs = {key: factory(**kwargs) for key, factory in self._factories.items()}
+        return self._class(**type_kwargs)  # type: ignore[call-arg]
+
+
+class DynamicTypeFactory(Factory):
+    """The template factory for dynamic Portex type.
+
+    Arguments:
+        decl: A dict indicate a dynamic Portex type.
```
Comment (Lee-000) [inline graviti/schema/factory.py:175]: Should the description in docstring be unified?

File: graviti/schema/factory.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+
+
+def string_factory_creator(decl: str) -> Union[VariableFactory, ConstantFactory[str]]:
+    """Check whether the input string is variable and returns the corresponding factory.
+
+    Arguments:
+        decl: A string which indicates a constant or a variable.
+
+    Returns:
+        A ``VariableFactory`` or a ``VariableFactory`` instance according to the input.
```
Comment (Lee-000) [inline graviti/schema/factory.py:408]:
```
A ``VariableFactory`` or a ``ConstantFactory``
```

**Review by graczhual — APPROVED** [review]

**Pull request merged**